### PR TITLE
fix: slow startup if no network is available

### LINKED ISSUE #5054: Painfully slow emulator start if no network is available
### [REQUIRED] Environment info

**firebase-tools:** 11.9.0

**Platform:** Windows

### [REQUIRED] Test case
Disable WIFI / unplug ethernet and start the emulator. 

### [REQUIRED] Steps to reproduce
Disable WIFI / unplug ethernet and start the emulator. You'll notice the emulator starts very very slowly because the `npm` command (almost) never returns.

https://github.com/firebase/firebase-tools/blob/9e524c18f4038dfcadbab637c9f661165feb1753/src/deploy/functions/runtimes/node/versioning.ts#L62

### [REQUIRED] Expected behavior
The emulator actually starts in a reasonable amount of time.

### [REQUIRED] Actual behavior
The emulator does not start in < 60s.

### PR DESCRIPTION
**Pull request by stfsy**
Preventing slow startups (> 60s) if no network is available. Tried to pick a reasonable timeout. Currently from Barcelona the NPM commands take 500ms to return. 

### Description
Fixes #5054

### CODE REVIEW

File: src/deploy/functions/runtimes/node/versioning.ts [hunk 1]
```diff
@@ -45,6 +46,7 @@ export function getFunctionsSDKVersion(sourceDir: string): string | void {
     const child = spawn.sync("npm", ["list", "firebase-functions", "--json=true"], {
       cwd: sourceDir,
       encoding: "utf8",
+      timeout: NPM_COMMAND_TIMEOUT_MILLIES,
```
Comment (colerogers) [inline src/deploy/functions/runtimes/node/versioning.ts:49]: We actually removed this spawn in a recent commit, do you mind pulling in the changes from master?

Comment (stfsy) [reply src/deploy/functions/runtimes/node/versioning.ts]: @colerogers thank you very much, rebased from master

File: src/deploy/functions/runtimes/node/versioning.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ interface NpmShowResult {
 }
 
 const MIN_SDK_VERSION = "2.0.0";
+const NPM_COMMAND_TIMEOUT_MILLIES = 3000;
```
Comment (colerogers) [inline src/deploy/functions/runtimes/node/versioning.ts:19]: We discussed this internally and think that `10000` would be a more appropriate timeout for customers with slower connectivity or an intermittent issue

Comment (stfsy) [reply src/deploy/functions/runtimes/node/versioning.ts]: @colerogers ok good point. I updated the value as per your recommendation

**Review by colerogers — APPROVED** [review]
Thanks for your contribution!

**Comment by stfsy** [comment]
@colerogers anything I need to? any more prerequisites for it getting merged? ☺

**Comment by stfsy** [comment]
@colerogers Thanks for merging!

**Pull request merged**